src/client.ts

```diff
@@ -1,191 +1,110 @@
-import { EventEmitter } from 'events';
-import { Peer, Request, WebSocketTransport, ProtooOptions } from 'protoo-client';
-import { v4 as uuidv4 } from 'uuid';
 import * as log from 'loglevel';
 
-import { LocalStream, RemoteStream, Stream } from './stream';
-import WebRTCTransport from './transport';
-import { TrackInfo, Notification } from './proto';
+import { LocalStream, RemoteStream } from './stream';
+import PeerConnection from './peerconnection';
 
 interface Config {
-  url: string;
-  uid?: string;
-  options?: ProtooOptions;
-  rtc?: RTCConfiguration;
-  loglevel?: log.LogLevelDesc;
+  rtc: RTCConfiguration;
+  loglevel: log.LogLevelDesc;
 }
 
-export default class Client extends EventEmitter {
-  dispatch: Peer;
-  uid: string;
-  rid: string | undefined;
-  localStreams: LocalStream[];
-  streams: { [name: string]: RemoteStream };
-  knownStreams: Map<string, Map<string, TrackInfo[]>>;
-
-  constructor(config: Config) {
-    super();
-    const uid = config.uid ? config.uid : uuidv4();
-
-    if (!config || !config.url) {
-      throw new Error('Undefined config or config.url in ion-sdk.');
-    }
+export default class Client {
+  private api: RTCDataChannel;
+  private pc: RTCPeerConnection;
+  private signal: Signal;
+  private candidates: RTCIceCandidateInit[];
 
-    const url = new URL(config.url);
-    url.searchParams.append('peer', uid);
-    const transport = new WebSocketTransport(url.toString(), config.options);
-    log.setLevel(config.loglevel !== undefined ? config.loglevel : log.levels.WARN);
+  ontrack?: (track: MediaStreamTrack, stream: RemoteStream) => void;
 
-    this.knownStreams = new Map();
-    this.uid = uid;
-    this.streams = {};
+  localStreams: LocalStream[];
+  remotes: { [name: string]: RemoteStream };
```

Comment:
Can we use Map<string,RemoteStream>?, I think it's more legible to remove and iterate elements